compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -556,6 +556,8 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
             args match {
               case arg :: Nil if isVarArg(arg) =>
                 addTyped(arg, formal)
+              case Typed(Literal(Constant(null)), _) :: Nil =>
+                addTyped(args.head, formal)
```

Comment:
Note that I used `.head` because `args` is  of type `List[Arg]` and
```scala
  case (arg @ Typed(Literal(Constant(null)), _)) :: Nil =>
    addTyped(arg, formal)
```
would fail with 
```scala
[error] 560 |                addTyped(arg, formal)
[error]     |                         ^^^
[error]     |Found:    dotty.tools.dotc.ast.Trees.Typed[dotty.tools.dotc.ast.Trees.Untyped](arg)
[error]     |Required: Arg
```

Reply:
What about just reject the code? We are moving towards null-safety, it seems it's better to reject. /cc: @abeln

Reply:
We should not always reject it. We could pass it to a Java varargs. Not sure if someone actually uses this pattern for performance reasons.

Reply:
In Java it is legal to pass `null` as the actual array of the varargs https://coderanch.com/t/631441/java/Varargs-null